src/core/jit/input_resolver.rs

```diff
@@ -66,4 +100,79 @@ where
             self.plan.is_introspection_query,
         ))
     }
+
+    #[allow(clippy::too_many_arguments)]
+    fn recursive_parse_arg(
+        &self,
+        parent_name: &str,
+        arg_name: &str,
+        type_of: &Type,
+        default_value: &Option<Output>,
+        value: Option<Output>,
+    ) -> Result<Option<Output>, ResolveInputError> {
+        let is_value_null = value.as_ref().map(|val| val.is_null()).unwrap_or(true);
+        let value: Result<Option<Output>, ResolveInputError> =
+            if !type_of.is_nullable() && value.is_none() {
+                let default_value = default_value.clone();
+                match default_value {
+                    Some(value) => Ok(Some(value)),
+                    None => Err(ResolveInputError::ArgumentIsRequired {
+                        arg_name: arg_name.to_string(),
+                        field_name: parent_name.to_string(),
+                    }),
+                }
+            } else if !type_of.is_nullable() && is_value_null {
```

Comment:
pile of matches is hard to read sometimes. Wdyt to use `let-else` and early returns instead, will it be more readable?